src/events.rs

```diff
@@ -8,12 +8,27 @@ use std::path::{Path, PathBuf};
 use std::pin::Pin;
 use std::task;
 
-use futures_util::pin_mut;
-use futures_util::stream::{Stream as FuturesStream, StreamExt};
+use futures_util::ready;
+use futures_util::stream::Stream as FuturesStream;
 use notify;
-use thiserror::Error;
 use tokio::sync::mpsc;
 
+type EventStream = mpsc::UnboundedReceiver<Result<notify::Event, notify::Error>>;
+
+fn notify_to_io_error(e: notify::Error) -> io::Error {
+    match e.kind {
+        notify::ErrorKind::Io(io_err) => io_err,
+        _ => {
+            // Runtime event errors should only be std::io, but
+            // need to handle this case anyway.
+            io::Error::new(
```

Comment:
Since `notify::Error` impls `std::error::Error`, you can just pass the error as-is to the `io::Error::new` function and maintain the error chain. I.e.

```rust
e => io::Error::new(io::ErrorKind::Other, e),
```

should work